dstep/main.d

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright: Copyright (c) 2016 Jacob Carlborg. All rights reserved.
+ * Authors: Jacob Carlborg
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+module dstep.main;
+
+import std.typecons : tuple, Tuple;
+import std.getopt;
+
+import dstep.Configuration;
+import dstep.translator.Options;
+import dstep.core.Exceptions;
+
+
+/**
+ *  Processes command-line arguments
+ *
+ *  Params:
+ *      args = command-line arguments
+ *
+ *  Returns:
+ *      2-element tuple with first element being aggregate struct of app
+ *      configuration and second - getopt parse result
+ */
+auto parseCLI (string[] args)
+{
+    import std.getopt;
+
+    Configuration config;
+
+    // Parse dstep own paramaters:
+
+    void parseLanguage (string param, string value)
+    {
+        config.clang_params ~= "-x";
+        config.clang_params ~= value;
+
+        switch (value)
+        {
+            case "c":
+            case "c-header":
+                config.language = Language.c;
+                break;
+            case "objective-c":
+            case "objective-c-header":
+                config.language = Language.objC;
+                break;
+            default:
+                throw new DStepException(`Unrecognized language "` ~ value ~ `"`);
+        }
+    }
+
+    bool forceObjectiveC;
+
+    auto helpInformation = getopt(
+        args,
+        std.getopt.config.passThrough,
+        std.getopt.config.caseSensitive,
+        "output|o", "Write output to", &config.output,
+        "language|x", "Treat subsequent input files as having type <language>", &parseLanguage,
+        "objective-c", "Treat source input file as Objective-C input.", &forceObjectiveC,
+        "no-comments", "Disable translation of comments", &config.no_comments
+    );
+
+    // remove dstep binary name (args[0])
+    args = args[1 .. $];
```

Comment:
Could you please check the different behavior of libclang when the name of the binary is included and when it's not. I think it might matter when libclang reports compile errors and warnings.